### DISCUSSION TITLE
How do I create a custom editor for executables (Unix)

### DISCUSSION THREAD

**Discussion by eugeneo** [microsoft/vscode-discussions#1532] [Extension Development QnA] [answered]
My team is working on an extension to show objdump output when an executable is opened in the editor.

The problem I am facing is how to assign custom editor to an executable. Executables on Unixes do not have set file extensions so glob patterns are not feasible.

**Comment by isidorn** [comment]
Maybe @lramos15 can help

**Comment by lramos15** [comment, marked as answer]
Your best bet would be to assign your custom editor to be the default binary editor and then have some sort of graceful fallback if it's not an executable.

`workbench.editor.defaultBinaryEditor`

<img width="677" alt="image" src="https://github.com/user-attachments/assets/b6b2c5ff-9cdf-47da-b4cc-247e0f9b9492">

**Reply by aharbii** [reply]
Providing CustomEditor prevents providing some functionality like using a CustomSymbolsProvider, or assigning a language to what we are displaying so we can use the syntax coloring for the language model. As we are mainly working with WebViews.

Also providing a CustomTextEditor will not solve the root issue of getting a warning when the user opens a binary file in a text editor.

It there any middle ground, where we could contribute a CustomEditor while keeping the functionality of providing symbols for this document? the syntax coloring part will not be an issue as it can be handled within the html.

**Reply by eugeneo** [reply]
@lramos For context - @aharbii is implementing an `objdump` display when opening an executable. Current implementation relies on the source editor facilities to provide highlight, code outline and such. 

This is his current implementation: https://github.com/uchen-ml/codeart-binexplore/pull/45

**Reply by aharbii** [reply]
[WebView Symbols](https://github.com/user-attachments/assets/18d44bba-feb6-4ab9-bffd-8d7e25c5ce59)

@lramos15 In this video you will be able to see the behavior I'm mentioning, when I open a custom webView editor, no symbols can be provided. 

It's more obvious in the markdown, when I open the markdown file with the text editor, I can see text document symbols.

**Reply by lramos15** [reply]
Do you want something like the document symbol provider to make your extension aware of the symbols? https://stackoverflow.com/questions/73823139/vscode-extension-using-executedocumentsymbolprovider-api

There is no way to keep the symbols view in a custom editor. Maybe a good feature request that you could make and see if the community supports is that if a user assigns a default binary editor they no longer see the warning.

**Reply by aharbii** [reply]
<img width="1440" alt="image" src="https://github.com/user-attachments/assets/5877ec2a-a100-47d7-8185-8251035341b3">

We already have implemented our own document symbols provider. For context, in this image, we parsed the binary with objdump, created a new text editor document, and provided the symbols for it. 

But yeah as you said 'There is no way to keep the symbols view in a custom editor', so the only option we have right now is to continue working with a text editor, not a custom editor.

I will try as you suggested and request the feature to bypass the warning. Thanks Logan!

**Reply by aharbii** [reply]
[Feature Request](https://github.com/microsoft/vscode/issues/230179)
